src/extension/background/vault-manager.ts

```diff
@@ -44,27 +48,38 @@ let vault: Vault = {
   ...defaultVault,
   encryptedSecretKey: localStorage.getItem(encryptedKeyIdentifier) || undefined,
   hasSetPassword: getHasSetPassword(),
+  salt: localStorage.getItem(saltIdentifier) || undefined,
 };
 
-export const getVault = () => {
-  return vault;
-};
+export const getVault = () => vault;
 
-export const vaultMessageHandler = async (message: MessageFromApp) => {
-  vault = await vaultReducer(message);
-  if (vault.encryptedSecretKey) {
-    localStorage.setItem(encryptedKeyIdentifier, vault.encryptedSecretKey);
+function persistOptional(storageKey: string, value?: string) {
+  if (value) {
+    localStorage.setItem(storageKey, value);
   } else {
-    localStorage.removeItem(encryptedKeyIdentifier);
+    localStorage.removeItem(storageKey);
   }
+}
+
+export async function vaultMessageHandler(message: MessageFromApp) {
+  vault = await vaultReducer(message);
+  persistOptional(encryptedKeyIdentifier, vault.encryptedSecretKey);
+  persistOptional(saltIdentifier, vault.salt);
   localStorage.setItem(hasSetPasswordIdentifier, JSON.stringify(vault.hasSetPassword));
   return vault;
-};
+}
+
+export function generateRandomHexString() {
+  const size = 16;
+  const randomValues = [...crypto.getRandomValues(new Uint8Array(size))];
+  return randomValues.map(val => ('00' + val.toString(16)).slice(-2)).join('');
+}
 
 async function storeSeed(secretKey: string, password?: string): Promise<Vault> {
+  const pw = password || DEFAULT_PASSWORD;
```

Comment:
default password 🤔  p4ssw0rd?

Reply:
Yeah, not the cleanest solution. `wallet-sdk` is built to assume there is always a password. The vault is designed to track if the user has actually entered a password. Otherwise, if the user exits the onboarding flow before they entered a password, they'll be sent back to the start of onboarding